src/testUtils.ts

```diff
@@ -126,6 +127,23 @@ export function extractInstanceTestName(symbolName: string): string {
 	return match[2];
 }
 
+/**
+ * Gets the appropriate debug arguments for a debug session on a test function.
+ * @param document  The document containing the tests
+ * @param testFunctionName The test function to get the debug args
+ * @param testFunctions The test functions found in the document
+ */
+export function getTestFunctionDebugArgs(document: vscode.TextDocument, testFunctionName: string, testFunctions: vscode.SymbolInformation[]): string[] {
+	const instanceMethod = extractInstanceTestName(testFunctionName);
+	if (instanceMethod) {
+		const testFns = findAllTestSuiteRuns(document, testFunctions);
```

Comment:
Would there ever be a case where `testFns` would be an empty array? i.e we weren't able to find the function corresponding to the suite.run()?

Reply:
yes it can ( e.g. if there is no suite method. but that will work as expected ( i.e. the regex wont match any suite since there is none). and the output will be a warning that no tests to run. 
![image](https://user-images.githubusercontent.com/16843371/51929003-167abb80-23ac-11e9-88ee-0d9dd058cd44.png)

Reply:
only thing that we might want to consider is on the CodeLens behavior in the following case:
- User writes a test as part of a test suite but doesn't add the test which runs the suite.  Which would be more helpful, not to show anything in codelens or to show them and when they run it to have the warning pasted above ( current behavior ) ?

Reply:
or we could try to return a custom error that no test suite exists ?

Reply:
> yes it can ( e.g. if there is no suite method. but that will work as expected ( i.e. the regex wont match any suite since there is none). and the output will be a warning that no tests to run.

In this case, we would be running `go test -test.run -testify.m CurrentTestFunctionName`
We know before hand this will not work. 

What would happen if we run `go test -test.run CurrentTestFunctionName` in such case?

Reply:
where `CurrentTestFunctionName` is a test suite test  same warning. Also FYI currently the same behavior is for running the tests as it was behavior.

Reply:
>  On that either way we will need to decide if want the warning when creating the codelens or when someone actually clicks it.

Warnings can be done either via a notification (prompts) or just printing it to the Dev Console.

The former is too intrusive for a task like creating codelens. Codelens are silent helpers and should not interfere with daily tasks of a user. 

The latter (Dev Console) is not too useful for the current case. A user will not be driven to open that up and read through unless there they feel that something is wrong.  In our case, the debug console would have given necessary information after they click on the codelens.

We can keep this as is.

Just one last piece of feedback.

The function `debugTestAtCursor` shares a lot of code with the function `testAtCursor`. Instead of having a new file for the former, let's move it to the same file as the latter. The common code between the 2 can be moved to `testUtils.ts` file

Reply:
@ramya-rao-a updated